src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -135,6 +135,22 @@ class DeltaTable private[tables](df: Dataset[Row], deltaLog: DeltaLog)
     executeHistory(deltaLog, None)
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Generate a manifest for the given Delta Table
+   *
+   * @param mode Specifies the mode for the generation of the manifest.
+   * See documentation online for the accepted mode values.
+   *
+   * @since 0.5.0
+   */
+  @Evolving
+  def generate(mode: String): Unit = {
```

Comment:
How is a new `mode` registered ?

Perhaps something like `generate(generator: Class[ManifestGenerator])` or `generate(generator: ManifestGenerator)`
 would make it more extendable and type safe..

Reply:
In general, your concern is right. It would be better to make it more type-safe for Scala. However, with our experience building Spark with multi-language APIs, it usually better if APIs are consistent across multiple languages. In Spark, that why we have chosen strings as the common way to take reader/writer options for example. We are trying to do the same here - have a consistent API across Scala/Python/SQL. 

That said, it is a fair point on the discoverability of the correct string is an issue. @rahulsmahadev we should add the valid string in the scala and python docs. As a list with the string at their explanation. Something like 
```
The valid modes are as follows (not case sensitive): 
- "symlink_manifest_format" - This will generate manifests in symlink format for Presto and Athena read support. See the online documentation for more information.
```
And in the future, if this become a usability issue for developers, we can always add additional scala-specific typesafe APIs in addition to this common APIs.

Reply:
Fair enough.
In that case I'd suggest a class to hold the constants instead..

```scala
DeltaTable.generate(ManifestFormat.SYMLINK_MANIFEST_FORMAT)
```

But what actually originated my comments here is the fact that as far as i can tell there is no way to register a new mode.. 
Is that intentional ?


I think it makes sense to make it somehow configurable.  
perhaps something like  :

```
spark.delta.manifestFormat.my_format=fully.qualified.ClassName
```

Of course this can be part of another patch, Just wanted to know if that is part of the plan..

Reply:
Oh, I understand now. No it was not intention of this API to provide a external way to register modes. That's is a whole different feature ... adding pluggability to add modes and commit hooks, etc. We haven't thought that far. :)

Reply:
Internally, the names are stored here - https://github.com/delta-io/delta/pull/262/files#diff-87fc704d0d12fc33dbef932679e1f587R64

If we have to add that class for scala, then we should add it for python as well. Let us think about it. In the mean time, let's merge this PR.